codice/ddf
# DDF-4163: Make WaitForReady a karaf command

#### What does this PR do?
- Created a SynchronisedInstaller to wait on different portions of the system to be ready
- Updated the installer to use the SynchronisedInstaller, resolves a race condition where the installer would continue through setup when not all the bundles were in an Active state
- Makes WaitForReady a karaf command instead of being in the start script
- Updates a portion of the itests to use the SynchronisedInstaller
- Disabled solr for itests that do not need it
- Made the solr port dynamic for a portion of the itests

#### Who is reviewing it? 
@figliold
@ethantmanns 
@jhunzik 
@rzwiefel 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@coyotesqrl

#### How should this be tested?
- Start up the distribution several times and ensure the wfr is waiting for all the bundles to reach an Active state.
- Go through the installer several times and ensure all the bundles reach an active state before the installer continues
- Review the logs of the SynchronisedInstaller and ensure they are accurately reflecting the state of the system

#### Any background context you want to provide?

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2182
https://codice.atlassian.net/browse/DDF-3483

#### Checklist:
- [In progress ] Documentation Updated
- [In progress] Update / Add Unit Tests
- [x ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
